### DISCUSSION TITLE
Has anyone tried using this plugin with the Spotify API?

### DISCUSSION THREAD

**Discussion by seannie-sharko** [grafana/grafana-infinity-datasource#546] [Q&A]
I've spent days trying to return my most current listened to tracks from my Spotify account but Infinity simply will not connect to Spotify's API using OAuth2

I created an app and passed through my client creds, along with scopes and token URL.

Endpoint:
https://api.spotify.com/v1/me/player/recently-played

Web API Reference:
https://developer.spotify.com/documentation/web-api/reference/#/operations/get-recently-played

Authorization Code Flow:
https://developer.spotify.com/documentation/general/guides/authorization/code-flow/

**Comment by yesoreyeram** [comment]
I haven't tried. But the code Authorization won't work as of now. What you need is client credentials workflow as described here https://developer.spotify.com/documentation/general/guides/authorization/client-credentials/. All the best.

**Reply by yesoreyeram** [reply]
one limitation is that you can't use autorized endpoints such as apis that include `/me/`?

<img width="857" alt="image" src="https://user-images.githubusercontent.com/153843/223192282-d2b9385d-207c-48b4-a02b-350c17a8e787.png">
